nexus/src/app/image.rs

```diff
@@ -356,26 +358,33 @@ impl super::Nexus {
         }
     }
 
-    // TODO-MVP: Implement
     pub(crate) async fn image_delete(
         self: &Arc<Self>,
         opctx: &OpContext,
         image_lookup: &ImageLookup<'_>,
     ) -> DeleteResult {
-        match image_lookup {
+        let image_param: sagas::image_delete::ImageParam = match image_lookup {
             ImageLookup::ProjectImage(lookup) => {
-                lookup.lookup_for(authz::Action::Delete).await?;
+                let (_, _, authz_image, image) =
+                    lookup.fetch_for(authz::Action::Delete).await?;
+                sagas::image_delete::ImageParam::Project { authz_image, image }
             }
             ImageLookup::SiloImage(lookup) => {
-                lookup.lookup_for(authz::Action::Delete).await?;
+                let (_, authz_image, image) =
+                    lookup.fetch_for(authz::Action::Delete).await?;
```

Comment:
These permissions checks seem to me the natural thing, but I want to make sure they actually work as you would expect, i.e., silo images would require stronger permissions to delete than project images. I believe the fact that `SiloImage` is a child of the silo but `ProjectImage` is a child of the project gets you all this for free. Unfortunately this is a little tedious to write tests around.

https://github.com/oxidecomputer/omicron/blob/463cc1ae256a5e22b9726ae58ea8b651e81e3882/nexus/db-queries/src/db/lookup.rs#L581-L588

https://github.com/oxidecomputer/omicron/blob/463cc1ae256a5e22b9726ae58ea8b651e81e3882/nexus/db-queries/src/db/lookup.rs#L659-L666

Reply:
I agree, I added a test in 2856f165a